### PR TITLE
Use the deck/spyglass URL to view logs on CI jobs

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

When Tekton CI job start, set the Tekton dashboard URL for the live
view of logs. When the CI job finishes, set the URL that points to
spyglass.

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

/kind feature

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: tekton/resources/cd/eventlistener.yaml [hunk 1]
```diff
@@ -169,6 +169,8 @@ spec:
               value:
                 - key: repo
                   expression: body.taskRun.metadata.annotations['tekton.dev/gitURL'].parseURL().path.substring(1)
+                - key: repoUnderscore
+                  expression: body.taskRun.metadata.annotations['tekton.dev/gitURL'].parseURL().path.substring(1).split('/').join('_')
```
Comment (afrittoli) [inline tekton/resources/cd/eventlistener.yaml:173]: Uhm, I've tried this with cel-eval and it works well, but I don't know if it's different in the context of an overlay, and the event listener did not seem happy about it

Comment (afrittoli) [reply tekton/resources/cd/eventlistener.yaml:173]:
```
{"level":"info","ts":"2022-07-04T17:02:01.408Z","logger":"eventlistener","caller":"sink/sink.go:299","msg":"interceptor stopped trigger processing: rpc error: code = InvalidArgument desc = error evaluating cel expression: expression \"body.taskRun.metadata.annotations['tekton.dev/gitURL'].parseURL().path.substring(1).split('/').join('_')\" check failed: ERROR: <input>:1:100: undeclared reference to 'join' (in container '')\n | body.taskRun.metadata.annotations['tekton.dev/gitURL'].parseURL().path.substring(1).split('/').join('_')\n | ...................................................................................................^","eventlistener":"tekton-events","namespace":"default","/triggers-eventid":"08f8e3ef-536a-47f3-af36-e1b29a2f6236","eventlistenerUID":"b37832bb-3e30-4516-9b04-c41d626e9321","/triggers-eventid":"08f8e3ef-536a-47f3-af36-e1b29a2f6236"}
```

Comment (afrittoli) [reply tekton/resources/cd/eventlistener.yaml:173]: This is new in triggers v0.20.x - so I need to deploy that in dogfooding first!

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**